src/pages/_app.tsx

```diff
@@ -13,22 +14,32 @@ const getLibrary = (provider: ExternalProvider | JsonRpcFetchFunc) =>
   new Web3Provider(provider)
 
 const App = ({ Component, pageProps }: AppProps): JSX.Element => (
-  <ChakraProvider theme={theme}>
-    <IconContext.Provider
-      value={{
-        color: "currentColor",
-        size: "1em",
-        weight: "bold",
-        mirrored: false,
-      }}
-    >
-      <Web3ReactProvider getLibrary={getLibrary}>
-        <Web3ConnectionManager>
-          <Component {...pageProps} />
-        </Web3ConnectionManager>
-      </Web3ReactProvider>
-    </IconContext.Provider>
-  </ChakraProvider>
+  <ChakraWrapper cookies={pageProps.cookies}>
```

Comment:
There're two ChakraProviders now inside each other. The theme object should be just passed to the one in ChakraWrapper. Also in the docs they define getServerSideProps inside the custom wrapper component, and just re-export that from here, I'd prefer to do that too